docs/store/datastore.md

```diff
@@ -112,6 +112,7 @@ authentication methods that use the `fsspec` mechanism.
 ### Google cloud storage
 * `GOOGLE_APPLICATION_CREDENTIALS` &mdash; path to the application credentials to use (in the form of a JSON file). This can
 be used if this file is located in a location on shared storage, accessible to pods executing MLRun jobs.
-* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this environment variable may contain
-the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
-and points `GOOGLE_APPLICATION_CREDENTIALS` at it.
\ No newline at end of file
+* `GCP_CREDENTIALS` &mdash; when the credentials file cannot be mounted to the pod, this secret or environment variable
+may contain the contents of this file. If configured in the function pod, MLRun dumps its contents to a temporary file 
+and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which uses `GCP_CREDENTIALS`
+directly.
```

Comment:
```suggestion
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS` contents
directly to the BigQuery client.
```

Reply:
Actually, it passes the contents to Spark.

Reply:
Ah, and to the client in pandas engine.

Reply:
```suggestion
and points `GOOGLE_APPLICATION_CREDENTIALS` at it. An exception is `BigQuerySource`, which passes `GCP_CREDENTIALS`'s
contents directly to the BigQuery client or, when using spark engine, to Spark.
directly.
```
?

Reply:
That works, though it may be a bit TMI - I'm fine with it.

Reply:
Could be "directly to the query engine". Wdyt?